core/src/com/unciv/logic/battle/Battle.kt

```diff
@@ -127,33 +129,83 @@ object Battle {
             } catch (ex: Exception) {
             } // parameter is not a stat
         }
-
+    }
+    
+    private fun tryCaptureUnit(attacker: ICombatant, defender: ICombatant) {
+        // https://forums.civfanatics.com/threads/prize-ships-for-land-units.650196/
+        // https://civilization.fandom.com/wiki/Module:Data/Civ5/GK/Defines
+        if (!defender.isDefeated()) return
+        if (attacker !is MapUnitCombatant) return
+        if (defender is MapUnitCombatant && !defender.getUnitType().isMilitary()) return
+        if (attacker.unit.getMatchingUniques("May capture killed [] units").none { defender.matchesCategory(it.params[0]) }) return
+        
+        var defendingStrength = 0
+        if (defender is CityCombatant) defendingStrength = defender.getCityStrength()
+        if (defender is MapUnitCombatant) defendingStrength = defender.unit.baseUnit.strength
+        
+        var captureChance = 10 + attacker.unit.baseUnit.strength.toFloat() / defendingStrength.toFloat() * 40
+        if (captureChance > 80) captureChance = 80f
+        if (100 * Random().nextFloat() > captureChance) return
+        
+        val newUnit = attacker.getCivInfo().placeUnitNearTile(defender.getTile().position, defender.getName())
+        if (newUnit == null) return // silently fail
+        attacker.getCivInfo().addNotification("Your [${attacker.getName()}] captured an enemy [${defender.getName()}]", newUnit.getTile().position, NotificationIcon.War)
+        
+        newUnit.currentMovement = 0f
+        newUnit.health = 50
     }
 
     private fun takeDamage(attacker: ICombatant, defender: ICombatant) {
-        var damageToDefender = BattleDamage.calculateDamageToDefender(attacker, attacker.getTile(), defender)
-        var damageToAttacker = BattleDamage.calculateDamageToAttacker(attacker, attacker.getTile(), defender)
+        var potentialDamageToDefender = BattleDamage.calculateDamageToDefender(attacker, attacker.getTile(), defender)
+        var potentialDamageToAttacker = BattleDamage.calculateDamageToAttacker(attacker, attacker.getTile(), defender)
+        
+        var damageDoneToAttacker = 0
```

Comment:
Instead of creating 2 new variables, just 'normalize' the value of the existing variables after combat to adjust for overkill.

Reply:
You said that before, but I don't understand what you mean. The variables get changed during the while loop, so to me it looks like the only way to determine what the total damage done is, is either to save their starting values and subtract the end value from it, or keep track of them during the while loop. Either that, or saving the starting health and subtract fhe ending health of both units. All of these options require extra variables